### PR TITLE
iOS device builds and watchman CI loop fix

### PR DESCRIPTION
**Pull request by HeyImChris**
## Summary

CI loops are running an unnecessary watchman script that is causing CI failures. Remove the script from our CI builds by creating 3 new targets to build instead. The yaml file will be updated to reflect that change in sdx-platform when this goes in. 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/Microsoft/react-native/pull/46)

### CODE REVIEW

File: RNTester/RNTester.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -191,6 +191,52 @@
 		9F465A02224EB106005BEE04 /* libRCTWebSocket.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 9FD21CF92242EFA400E7F88E /* libRCTWebSocket.a */; };
 		9F465A03224EB10E005BEE04 /* libReact.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 9FD21D122242EFA400E7F88E /* libReact.a */; };
 		9F465A05224EB177005BEE04 /* libRCTTest-macOS.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 18DF57271EFA056600BF4666 /* libRCTTest-macOS.a */; };
+		9F6168B9225EA318008F89F1 /* libRCTWebSocket.a in Frameworks */ = {isa = PBXBuildFile; fileRef = 139FDED91B0651EA00C62182 /* libRCTWebSocket.a */; };
```
Comment (markavitale) [inline RNTester/RNTester.xcodeproj/project.pbxproj]: Maybe I'm misreading this, but why would static libs (.a's) go in the frameworks folder?

Comment (HeyImChris) [reply RNTester/RNTester.xcodeproj/project.pbxproj]: That's what Xcode calls libraries/frameworks in the "Link Binary with Libraries" build phase. See attached image and note the label at the bottom

Screen Shot 2019-04-12 at 10 25 46 AM

**Comment by markavitale** [comment]
> By mirroring the 3 existing flavors (iphone device, iphone simulator, mac) in CI only targets, we can link in all the libraries we need to, but not run that watchman script that takes of time and also fails. Tom already created one of those flavors (iphone device) to handle some code signing issue and I'm reusing that which is why you only see two added.

Like we discussed in person, this may make sense to follow up and deduplicate the build logic between the "local" targets and the "cloud build" targets, but let's get this in to unblock cloud builds.

**Pull request merged**